src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/PullRequestStoreImpl.java

```diff
@@ -40,11 +64,29 @@ public boolean hasOpenPullRequests(String branchName, BitbucketSCMRepository rep
                 .isPresent();
     }
 
+    /**
+     * retrieves a pull request given ids and keys
+     * @param key
+     * @param slug
+     * @param serverId
+     * @param requestId
+     * @return desired pull request else null
+     */
     @Override
-    public ConcurrentMap<PullRequestStoreImpl.CacheKey, ConcurrentLinkedQueue<BitbucketPullRequest>> getPullRequests() {
-        return pullRequests;
+    public BitbucketPullRequest getPullRequest(String key, String slug, String serverId, int requestId) {
```

Comment:
As mentioned during pairing, I think we should change this to return an `Optional<BitbucketPullReqest>` as that makes it adhere to the non-null return and forces callers to handle the case of not finding the PR they were looking for.